libs/acn/RPTPDU.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * RPTPDU.cpp
+ * The E1.33 RPT PDU
+ * Copyright (C) 2024 Peter Newman
+ */
+
+
+#include "ola/Logging.h"
+#include "ola/base/Array.h"
+#include "ola/network/NetworkUtils.h"
+#include "libs/acn/RPTPDU.h"
+
+namespace ola {
+namespace acn {
+
+using ola::io::OutputStream;
+using ola::network::HostToNetwork;
+
+/*
+ * Size of the header portion.
+ */
+unsigned int RPTPDU::HeaderSize() const {
+  return sizeof(RPTHeader::rpt_pdu_header);
+}
+
+
+/*
+ * Size of the data portion
+ */
+unsigned int RPTPDU::DataSize() const {
+  return m_pdu ? m_pdu->Size() : 0;
+}
+
+
+/*
+ * Pack the header portion.
+ */
+bool RPTPDU::PackHeader(uint8_t *data, unsigned int *length) const {
+  unsigned int header_size = HeaderSize();
+
+  if (*length < header_size) {
```

Comment:
would this be a null-dereference if someone forgets to pass length? Should check for `!= nullptr` before?

Reply:
It does indeed segfault. I've now added something a bit higher up in the stack which covers us for everything apart from someone calling PackHeader/PackData directly themselves, but means we don't need to tweak every single PDU...

Reply:
Where's the commit for those changes "higher up in the stack"? I've seen your changes but it looks like they are per PDU?

Reply:
https://github.com/OpenLightingProject/ola/pull/1950/commits/d6a13dd9218c4f335905318b200462ac2218ed2d

There's basically no uses of PackHeader/PackData directly, only via Pack, hence just doing it there for now.

I've also added a proper fix to the backlog in #1841 as there are a lot of other bits to cover too.